packages/math-input/src/components/keypad/__tests__/keypad-v2-mathquill.test.tsx

```diff
@@ -231,4 +247,45 @@ describe("Keypad v2 with MathQuill", () => {
 
         expect(mockMathInputCallback).toHaveBeenLastCalledWith("");
     });
+
+    // Keypad event tests
+    it("fires the keypad open event on open", () => {
+        // Arrange
+        const mockSendEvent = jest.fn();
+        render(
+            <V2KeypadWithMathquill
+                onChangeMathInput={() => {}}
+                keypadClosed={true}
+                sendEvent={(event) => mockSendEvent(event.type)}
+            />,
+        );
+
+        // Act
+        userEvent.click(screen.getByRole("button", {name: "Keypad toggle"}));
+
+        // Assert
+        expect(mockSendEvent).toHaveBeenLastCalledWith(
+            "math-input:keypad-opened",
+        );
+    });
+
+    // Keypad event tests
+    it("fires the keypad open event on close", () => {
+        // Arrange
+        const mockSendEvent = jest.fn();
+        render(
+            <V2KeypadWithMathquill
+                onChangeMathInput={() => {}}
+                sendEvent={(event) => mockSendEvent(event.type)}
```

Comment:
Same comment here as above.  Let's pass the mock in directly:

```suggestion
                sendEvent={mockSendEvent}
```